x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/CreateIndexFromSourceTransportAction.java

```diff
@@ -89,9 +89,12 @@ protected void doExecute(Task task, CreateIndexFromSourceAction.Request request,
 
         Settings.Builder settings = Settings.builder()
             // first settings from source index
-            .put(filterSettings(sourceIndex))
-            // then override with request settings
-            .put(request.settingsOverride());
+            .put(filterSettings(sourceIndex));
+
+        if (request.settingsOverride().isEmpty() == false) {
```

Comment:
Normally this would be overkill bit the `keySet()` operation is non trivial for the `Settings` object so this may save a few cycles